File: processor/annotations/src/main/java/com/google/appinventor/components/common/OptionList.java
package com.google.appinventor.components.common;

/**
 * A marker interface for defining option list helper blocks via the Java scripting language. This can be expanded later
 * if we want to get more information out of option list definitions.
 */
public interface OptionList<T> {
    /**
     * Returns the underlying value of the option.
     *
     * @return the underlying value of the option
     */
    T toUnderlyingValue();
}

File: processor/annotations/src/main/kotlin/com/google/appinventor/components/common/OptionList.kt
package com.google.appinventor.components.common

/**
 * A marker interface for defining option list helper blocks via the Java scripting language.
 * This can be expanded later if we want to get more information out of option list definitions.
 */
interface OptionList<T> {
    fun toUnderlyingValue(): T
}